internal/datastore/common/cursor.go

```diff
@@ -0,0 +1,49 @@
+package common
+
+import (
+	"encoding/base64"
+	"fmt"
+
+	"github.com/authzed/spicedb/pkg/datastore"
+	"github.com/authzed/spicedb/pkg/datastore/options"
+	core "github.com/authzed/spicedb/pkg/proto/core/v1"
+)
+
+const (
+	errEncodingCursor = "error encoding cursor: %s"
+	errDecodingCursor = "error decoding cursor: %s"
+)
+
+func NoopCursorFunc(_ *core.RelationTuple) (options.Cursor, error) {
+	return "", datastore.ErrCursorsWithoutSorting
+}
+
+func EncodedProtoCursor(tpl *core.RelationTuple) (options.Cursor, error) {
+	if tpl == nil {
+		return "", datastore.ErrCursorEmpty
+	}
+
+	marshalled, err := tpl.MarshalVT()
+	if err != nil {
+		return "", fmt.Errorf(errEncodingCursor, err)
+	}
+	return options.Cursor(base64.StdEncoding.EncodeToString(marshalled)), nil
```

Comment:
Are these cursors ever sent outside of internal code? If so, and we ever have to change this format, we might want to consider using an encoded proto with a version, rather than just the tuple itself. If not, then no worries

Reply:
Changed cursors to just be a `RelationTuple` pointer.